### ISSUE TITLE
Fatal Error of tf.softmax in webgl mode

### ISSUE THREAD

**Issue by weishengkui** [tensorflow/tfjs#5394]
FATAL ERROR, tf.softmax may compute incorrect output when webgl mode is set, such as tf.setBackend('webgl');

Before softmax.
Tensor
```
    [[[[-0.0293463, -1000000000, -1000000000, ..., -1000000000, -1000000000, -1000000000],
       [-0.2220541, 0.231505   , -1000000000, ..., -1000000000, -1000000000, -1000000000],
       [0.0011374 , -0.0965061 , -0.0123992 , ..., -1000000000, -1000000000, -1000000000],
       ...,
       [0.0958247 , 0.0556024  , 0.1016628  , ..., -0.0275311 , -1000000000, -1000000000],
       [-0.0398366, 0.0641379  , 0.1550138  , ..., -0.1018429 , -0.216876  , -1000000000],
       [0.259409  , -0.5869865 , 0.1881166  , ..., 0.4390099  , -0.0278636 , 0.0095359  ]],
```

After softmax...

print.js:34 Tensor

```
    [[[[3.4028234663852886e+38, 3.4028234663852886e+38, 3.4028234663852886e+38, ..., 3.4028234663852886e+38, 3.4028234663852886e+38, 3.4028234663852886e+38],
       [0.3885149             , 0.6114851             , 0                     , ..., 0                     , 0                     , 0                     ],
       [0.345599              , 0.3134487             , 0.3409523             , ..., 0                     , 0                     , 0                     ],
       ...,
       [0.0180443             , 0.0173329             , 0.01815               , ..., 0.0159502             , 0                     , 0                     ],
```

**Comment by rthadur** [comment]
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by weishengkui** [comment]
Here is code for debugging:

`    tf.setBackend('webgl');
    // tf.setBackend('cpu');
    const dd = tf.randomNormal([32, 4, 64, 64], 0, 1, "float32");
    dd.print();
    const soft_result = tf.softmax(dd);
    soft_result.print();`

**Comment by weishengkui** [comment]
![image](https://user-images.githubusercontent.com/23742600/127448423-650fce1f-ba34-4ea9-8862-1b5386a2d9b7.png)

**Comment by raffizulvian** [comment]
You are generating a random tensor without passing the `seed` parameter to `tf.randomNormal`. So every time you run the code, the elements of the tensor produced will be different from the previous run. This is why the softmax function gives different outputs.

To solve the issue just give `tf.randomNormal` an integer as argument for this function `seed` parameter.
```js
tf.setBackend('webgl');
// tf.setBackend('cpu');

const dd = tf.randomNormal([32, 4, 64, 64], 0, 1, "float32", 90);  // I use 90 as the seed
// dd.print();

const soft_result = tf.softmax(dd);
soft_result.print();
```

First run using CPU backend:
![image](https://user-images.githubusercontent.com/60030585/158010691-0b24559b-3736-4997-aa81-fb3bc2a4c962.png)

Second run using WebGL backend:
![image](https://user-images.githubusercontent.com/60030585/158010724-b5f00958-bbb7-4517-81a0-8b4cd99c240a.png)

Notice that in both runs, the result stays the same even though I change the backend.

Hope this helps @weishengkui !
